### PR TITLE
Adds hair gradient to DNA

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Adds hair gradient to DNA information to DNA, allowing to to persist through cloning and changeling transformation.
Fixes: #22990

## Why It's Good For The Game
Hair gradients should be retained during transfer/transformation of a body.

## Testing
Give myself a weird gradient with an offset.
Hop in the cloner and die.
Get cloned and have a hair gradient.

## Changelog
:cl:
tweak: Hair gradient is now also stored in DNA
fix: Hair gradient is transferred during a transformation or cloning.
/:cl:

### CODE REVIEW

**Comment by S34NW** [comment]
GA's are gonna need sufficient notice of this going in so they can update their characters

File: code/__DEFINES/dna.dm [hunk 1]
```diff
@@ -41,18 +41,24 @@
...
-#define DNA_UI_GENDER		32
-#define DNA_UI_BEARD_STYLE	33
-#define DNA_UI_HAIR_STYLE	34
-#define DNA_UI_HACC_STYLE	35
-#define DNA_UI_BACC_STYLE	36
-#define DNA_UI_HEAD_MARK_STYLE	37
-#define DNA_UI_BODY_MARK_STYLE	38
-#define DNA_UI_TAIL_MARK_STYLE	39
-#define DNA_UI_LENGTH		40 // Update this when you add something, or you WILL break shit.
+#define DNA_UI_GRADIENT_R	29
```
Comment (Contrabang) [inline code/__DEFINES/dna.dm:44]: I think these should be DNA_UI_HAIR_GRADIENT_..., because otherwise they could be confused a gradient for something else

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/dna/dna2_helpers.dm [hunk 2]
```diff
@@ -204,6 +204,14 @@
 	head_organ.hair_colour = rgb(head_organ.dna.GetUIValueRange(DNA_UI_HAIR_R, 255), head_organ.dna.GetUIValueRange(DNA_UI_HAIR_G, 255), head_organ.dna.GetUIValueRange(DNA_UI_HAIR_B, 255))
 	head_organ.sec_hair_colour = rgb(head_organ.dna.GetUIValueRange(DNA_UI_HAIR2_R, 255), head_organ.dna.GetUIValueRange(DNA_UI_HAIR2_G, 255), head_organ.dna.GetUIValueRange(DNA_UI_HAIR2_B, 255))
 
+	//Hair Gradient
+	var/gradient = GetUIValueRange(DNA_UI_HAIR_GRADIENT_STYLE,GLOB.hair_gradients_list.len)
```
Comment (Contrabang) [inline code/game/dna/dna2_helpers.dm:208]:
```suggestion
	var/gradient = GetUIValueRange(DNA_UI_HAIR_GRADIENT_STYLE, length(GLOB.hair_gradients_list))
```

File: code/game/dna/dna2_helpers.dm [hunk 3]
```diff
@@ -204,6 +204,14 @@
...
+	if((gradient > 0) && (gradient <= length(GLOB.hair_gradients_list)))
```
Comment (Contrabang) [inline code/game/dna/dna2_helpers.dm:209]:
```suggestion
	if(ISINDEXSAFE(GLOB.hair_gradients_list, gradient))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**